OC-4074 Update User

hosh opened the pull request:
PUT /users/:name
- Updates admin flag
- Regenerates key pair
- Does NOT update password
- Does NOT check authorization

hosh committed:
[OC-4074] Added chef_wm_named_user:from_json()

hosh committed:
[OC-4074] Cleanup comments

hosh committed:
[OC-4074] PUT /users/:name with private_key set to true should regenerate key pair

hosh committed:
[OC-4074] PUT /users/<name> should accept password changes

seth commented on src/chef_wm_base.erl:
```diff
@@ -299,6 +299,7 @@ update_from_json(#wm_reqdata{} = Req, #base_state{chef_db_context = DbContext,
                                                   requestor_id = ActorId}=State,
                  OrigObjectRec, ObjectEjson) ->
     ObjectRec = chef_object:update_from_ejson(OrigObjectRec, ObjectEjson),
+
```
Would make sense to remove this change since it is whitespace only.

hosh committed:
[OC-4074] Refactored user update handling

- Send password out of band
- Send keypair through ejson
- Use custom update_from_ejson handler

hosh committed:
[OC-4074] Use custome chef_wm_named_user:update_from_json()

- chef_wm_named_user:update_from_json() should call chef_user:update_from_ejson() directly
- Removed code pertaining to solr indexing for user. (Users are not
  indexed anyways)

seth commented:
local tests pass and pedant run for me. Merge when ready
